kernel/comps/framebuffer/src/ansi_escape.rs

```diff
@@ -161,17 +161,75 @@ impl EscapeFsm {
                 }
 
                 // Set foreground and background colors
-                // Reference: <https://en.wikipedia.org/wiki/ANSI_escape_code#3-bit_and_4-bit>
+                // Reference: <https://en.wikipedia.org/wiki/ANSI_escape_code#Colors>
                 30..=37 => op.set_fg_color(COLORS[*param as usize - 30]),
+                38 => {
+                    if num_params == 3 && self.params[1] == 5 {
```

Comment:
This should be simplified as:
```rust
                38 if num_params == 3 && self.params[1] == 5 => {
```

The same applies to other `if` statements inserted in similar places.